src/watch.rs

```diff
@@ -26,98 +27,122 @@ where
 {
     println!("Watching for changes...");
 
-    let rx_changes = trigger_on_change(&mut watch_sources, &mut watch_code)?;
+    let mut watch_sources_old: HashSet<_> = watch_sources.collect();
+    let mut watch_code_old: HashSet<_> = watch_code.collect();
+
+    trigger_on_change(
+        watch_sources_old.clone(),
+        watch_code_old.clone(),
+        |change, sw, cw| {
```

Comment:
One thing I just noticed: You neither need the closure API nor to keep the watchers on the stacks of the background threads, you could just return `(change, sw, cw)` from `trigger_on_change`.

Reply:
with `change`, you mean the receiver?

Reply:
Yes, just returning a tuple of the tree things you pass arguments to the closure.